src/coreclr/gc/unix/configure.cmake

```diff
@@ -184,4 +184,21 @@ check_prototype_definition(
     ${STATFS_INCLUDES}
     HAVE_NON_LEGACY_STATFS)
 
+set(CMAKE_REQUIRED_LIBRARIES)
+check_cxx_source_runs("
+#include <fcntl.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+
+int main(void) {
+  int fd;
+
+  fd = open(\"/proc/self/statm\", O_RDONLY);
+  if (fd == -1) {
+    exit(1);
+  }
+  exit(0);
+}" HAVE_PROCFS_STATM)
```

Comment:
What's the point of checking this at build time, shouldn't the runtime not depends on whether the build machine exposes statm?